domain/miningmanager/mempool/mempool.go

```diff
@@ -39,42 +43,90 @@ func (mp *mempool) ValidateAndInsertTransaction(transaction *externalapi.DomainT
 	mp.mtx.Lock()
 	defer mp.mtx.Unlock()
 
-	return mp.validateAndInsertTransaction(transaction, isHighPriority, allowOrphan)
+	return mp.validateAndInsertTransaction(transaction, isHighPriority, allowOrphan, true)
 }
 
-func (mp *mempool) GetTransaction(transactionID *externalapi.DomainTransactionID) (*externalapi.DomainTransaction, bool) {
+func (mp *mempool) GetTransaction(transactionID *externalapi.DomainTransactionID,
+	includeTransactionPool bool,
+	includeOrphanPool bool) (
+	transactionPoolTransaction *externalapi.DomainTransaction,
+	orphanPoolTransaction *externalapi.DomainTransaction,
+	found bool) {
+
 	mp.mtx.RLock()
 	defer mp.mtx.RUnlock()
 
-	return mp.transactionsPool.getTransaction(transactionID)
-}
+	var transactionfound bool
+	var orphanTransactionFound bool
 
-func (mp *mempool) AllTransactions() []*externalapi.DomainTransaction {
-	mp.mtx.RLock()
-	defer mp.mtx.RUnlock()
+	if includeTransactionPool {
+		transactionPoolTransaction, transactionfound = mp.transactionsPool.getTransaction(transactionID, true)
+	}
+	if includeOrphanPool {
+		orphanPoolTransaction, orphanTransactionFound = mp.orphansPool.getOrphanTransaction(transactionID, true)
```

Comment:
I think i did this for testing purposes (before this pr we had get orphan and get transaction as two seperate calls, it ended so that the rpc calls had both orphans and transaction pool transactions), ideally it shouldn't happen under one lock, i only see a slight usage case for testing the mempool. but will remove it, since it has no functional purpose.